File: Rotate Array/Rotate_Array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int[] newNums = new int[nums.length];
        for (int i = 0; i < nums.length; i++){
            int newIndex = i + k;
            if (newIndex >= nums.length){
                newIndex = newIndex % nums.length;
            }
            newNums[newIndex] = nums[i];
        }
        for (int i = 0; i < nums.length; i++){
            nums[i] = newNums[i];
        }
    }
}

File: Coding Problems & My Solutions/Rotate Array/Rotate_Array.swift
class Solution {
    func rotate(_ nums: inout [Int], _ k: Int) {
        var k = k
        if k >= nums.count {
            k = k % nums.count
        }
        if k < 1 {
            return
        }
        var rightArr: [Int] = []
        for i in 0..<nums.count-k {
            rightArr.append(nums[i])
        }
        var leftArr: [Int] = []
        for i in nums.count-k..<nums.count {
            leftArr.append(nums[i])
        }
        nums = leftArr + rightArr
    }
}